### ISSUE TITLE
--crypt-directory-name-encryption not parsed / breaks rclone

### ISSUE THREAD

**Issue by hyperknot** [ncw/rclone#2860]
#### What is the problem you are having with rclone?
Using `--crypt-directory-name-encryption` breaks rclone. If I remove that single argument from my command below, it works correctly.

#### What is your rclone version (output from `rclone version`)
```
rclone v1.45
- os/arch: darwin/amd64
- go version: go1.11.2
```

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
macOS 10.13.6

####  Which cloud storage system are you using? (eg Google Drive)
Backblaze B2

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
I'm trying to run crypted sync, without using config file.

```
password=`rclone obscure x`

rclone sync \
    --b2-account a \
    --b2-key b \
    --crypt-remote :b2:c \
    --crypt-filename-encryption off \
    --crypt-directory-name-encryption false \
    --crypt-password "$password" \
    -vv \
    . :crypt:
```

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)
```
2018/12/23 01:54:33 DEBUG : rclone: Version "v1.45" starting with parameters ["rclone" "sync" "--b2-account" "a" "--b2-key" "b" "--crypt-remote" ":b2:c" "--crypt-filename-encryption" "off" "--crypt-directory-name-encryption" "false" "--crypt-password" "LpijcJx8ESN58IspI8vCri8" "-vv" "." ":crypt:"]
Usage:
  rclone sync source:path dest:path [flags]

Flags:
  -h, --help   help for sync

Use "rclone [command] --help" for more information about a command.
Use "rclone help flags" for to see the global flags.
Use "rclone help backends" for a list of supported services.
Command sync needs 2 arguments maximum
```

**Comment by ncw** [comment]
What you want is `--crypt-directory-name-encryption=false` for boolean flags you can make them true with `--flag=true` or `--flag` but you need `--flag=false` to make them false.  They don't accept the `--flag true` syntax though.

This should probably be explained better in the docs though!

**Comment by hyperknot** [comment]
Thanks, yes, I figured it out that this one flag needs to be used with `=`. I guess the behaviour is good as it is, but the error message doesn't help at all (I had to manually comment out individual segments), as well as an explanation would be very nice in the documentations!

**Comment by ncw** [comment]
> Thanks, yes, I figured it out that this one flag needs to be used with `=`.

Great!

> I guess the behaviour is good as it is, but the biggest problems is the error message, which doesn't help at all

What happened is that  `--crypt-directory-name-encryption` got parsed as a boolean flag which left its argument `false` as another parameter - hence the error message.  If I wrote the arguments in the error message that would at least give you a clue.

I've improved the error message slightly

    Command sync needs 2 arguments maximum: you provided 3 non flag arguments: ["false" "." ":crypt:"]

I've also added a bit of text to the docs

Options that take parameters can have the values passed in two ways, `--option=value` or `--option value`. However boolean (true/false) options behave slightly differently to the other options in that
`--boolean` sets the option to `true` and the absence of the flag sets it to `false`.  It is also possible to specify `--boolean=false` or `--boolean=true`.  Note that `--boolean false` is not valid - this is parsed as `--boolean` and the `false` is parsed as an extra command line argument for rclone.

Which should hopefully address your concerns :-)

> Otherwise, thanks for this amazing and stable piece of software, I'm migrating more and more vendor specific CLIs to rclone and it's working really well!

You are welcome - and great - that is exactly what rclone is for!  It is really useful to have good feedback for people using rclone in different ways so thank you for that.
